rasa/utils/tensorflow/models.py

```diff
@@ -881,11 +885,11 @@ def _get_batch_dim(attribute_data: Dict[Text, List[tf.Tensor]]) -> int:
         # All the values in the attribute_data dict should be lists of tensors, each
         # tensor of the shape (batch_dim, ...). So we take the first non-empty list we
         # encounter and infer the batch size from its first tensor.
-        for key, data in attribute_data.items():
+        for data in attribute_data.values():
             if data:
                 return tf.shape(data[0])[0]
 
-        return 0
+        return None
```

Comment:
Shouldn't the return type change to `Optional[int]` then?

Reply:
Ah thanks for spotting this! I tried to revert some None/0 merges from main and this one went wrong.

Reply:
won't change this - types will be fixed by changes in main